examples/non_argument_deps_successor/non_argument_deps_successor/assets.py

```diff
@@ -0,0 +1,21 @@
+from dagster import AssetOut, asset, multi_asset
+
+
+@asset
```

Comment:
this minimal example gets the dependency structure correct and materializes successfully 
![Screenshot 2023-06-21 at 2 08 37 PM](https://github.com/dagster-io/dagster/assets/19783112/74aba3a7-3721-443c-80e3-a0af9edfe7b7)